master/lustregittagpoller.py

```diff
@@ -0,0 +1,122 @@
+# -*- python -*-
+# ex: set syntax=python:
+
+import itertools
+import os
+
+from buildbot import config
+from twisted.internet import defer
+from twisted.internet import utils
+from twisted.python import log
+from buildbot.util import epoch2datetime
+from buildbot.changes.gitpoller import GitPoller 
+
+class LustreTagPoller(GitPoller):
```

Comment:
Since you are making the "branch" and "branches" arguments invalid in this class, you will also need to override the parent class's describe() method with your own in this subclass.  Look through the parent class for other methods that need to be replaced to make sense in this subclass.